### ISSUE TITLE
Valid whitespace characters in feature files?

### ISSUE THREAD

**Issue by jenskutilek** [adobe-type-tools/afdko#191]
`makeotf` apparently allows non-breaking space characters as whitespace in feature files, while `fontTools.feaLib` doesn’t. Is there a list of which whitespace characters are valid somewhere?

**Comment by frankrolf** [comment]
In my opinion, all whitespace characters should be valid in feature files (even line breaks).
I have been teaching the concept that the  only requirement is a semicolon to end a logical “line”. 

This should be valid:
```
@EXAMPLE = [
    A
    B
    C
];
```

**Comment by khaledhosny** [comment]
Whitespace characters is a long list in Unicode (and potentially open ended with future additions to Unicode), the spec should explicitly list what characters are treated as such instead.

**Comment by frankrolf** [comment]
Since the feature file specification does not allow for non-ASCII characters, I don’t think this is an issue.

**Comment by kenlunde** [comment]
The non-breaking space is U+00A0, which is (barely) outside of ASCII, so based on @frankrolf's statement, its presence outside of comments should throw an error.

While we're on the subject, the following are the [Unicode Version 10.0](http://www.unicode.org/Public/10.0.0/ucd/PropList.txt) characters that have the _White_Space_ property:

```0009..000D    ; White_Space # Cc   [5] <control-0009>..<control-000D>
0020          ; White_Space # Zs       SPACE
0085          ; White_Space # Cc       <control-0085>
00A0          ; White_Space # Zs       NO-BREAK SPACE
1680          ; White_Space # Zs       OGHAM SPACE MARK
2000..200A    ; White_Space # Zs  [11] EN QUAD..HAIR SPACE
2028          ; White_Space # Zl       LINE SEPARATOR
2029          ; White_Space # Zp       PARAGRAPH SEPARATOR
202F          ; White_Space # Zs       NARROW NO-BREAK SPACE
205F          ; White_Space # Zs       MEDIUM MATHEMATICAL SPACE
3000          ; White_Space # Zs       IDEOGRAPHIC SPACE```

**Comment by jenskutilek** [comment]
Yes, 0x0a is not ASCII, but some font editors apparently do not prevent non-ASCII characters from ending up in feature files. For example, in my case the file was in a UFO generated by glyphsLib from a Glyphs file. The feature file was UTF-8-encoded. Maybe this should be decided together with #165.
